File: evaluation/benchmark/matmult.py
#!/usr/bin/python

import sys

def createMatrix(height,width,val):
  matrix = [None]*height
  for y in range(0,height):
    matrix[y] = [0]*width
  for y in range(0,height):
    for x in range(0,width):
      matrix[y][x] = val%23
      val += 1
  return matrix

def printMatrix(matrix):
  height = len(matrix)
  width = len(matrix[0])

  for y in range(0,height):
    line = ""
    for x in range(0,height):
      line += "%5d "%(matrix[y][x])
    print line

def multiply(a,b):
  arows = len(a)
  acols = len(a[0])
  bcols = len(b[0])

  result = [None]*arows
  for y in range(0,arows):
    result[y] = [0]*bcols

  for y in range(0,arows):
    for x in range(0,bcols):
      for i in range(0,acols):
        result[y][x] += a[y][i] * b[i][x]

  return result

def matsum(matrix):
  height = len(matrix)
  width = len(matrix[0])
  total = 0
  for y in range(0,height):
    for x in range(0,width):
      total += matrix[y][x]
  return total

size = 10;
if (len(sys.argv) > 1):
  size = int(sys.argv[1])
doprint = (len(sys.argv) != 2)

a = createMatrix(size,size,1)
b = createMatrix(size,size,2)
if (doprint):
  printMatrix(a);
  print("--")
  printMatrix(b);
  print("--")

c = multiply(a,b);
if (doprint):
  printMatrix(c);
  print("");

print("sum = "+str(matsum(c)));

File: evaluation/benchmark/matmult.js
function createMatrix(height,width,val)
{
  var matrix = new Array(height);
  for (var y = 0; y < height; y++)
    matrix[y] = new Array(width);
  for (var y = 0; y < height; y++) {
    for (var x = 0; x < width; x++) {
      matrix[y][x] = val%23;
      val++;
    }
  }
  return matrix;
}

function printMatrix(matrix)
{
  var height = matrix.length;
  var width = matrix[0].length;
  for (var y = 0; y < height; y++) {
    var line = "";
    for (var x = 0; x < width; x++) {
      var str = ""+matrix[y][x];
      for (var i = 0; i < 5-str.length; i++)
        line += " ";
      line += str;
      line += " ";
    }
    print(line);
  }
}

function multiply(a,b)
{
  var arows = a.length;
  var acols = a[0].length;
  var bcols = b[0].length;

  var result = new Array(arows);
  for (var y = 0; y < arows; y++)
    result[y] = new Array(bcols);

  for (var y = 0; y < arows; y++) {
    for (var x = 0; x < bcols; x++) {
      result[y][x] = 0;
      for (var i = 0; i < acols; i++)
        result[y][x] += a[y][i] * b[i][x];
    }
  }

  return result;
}

function matsum(matrix)
{
  var total = 0;
  for (var y = 0; y < matrix.length; y++)
    for (var x = 0; x < matrix[0].length; x++)
      total += matrix[y][x];
  return total;
}

var size = 10;
if (arguments.length > 0)
  size = arguments[0];
var doprint = (arguments.length != 1);

var a = createMatrix(size,size,1);
var b = createMatrix(size,size,2);
if (doprint) {
  printMatrix(a);
  print("--");
  printMatrix(b);
  print("--");
}

var c = multiply(a,b);
if (doprint) {
  printMatrix(c);
  print("");
}
print("sum = "+matsum(c));
